File: src/main/java/com/vaadin/demo/component/numberfield/NumberFieldReadonlyAndDisabled.java
package com.vaadin.demo.component.numberfield;

import com.vaadin.flow.component.formlayout.FormLayout;
import com.vaadin.flow.component.textfield.NumberField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("number-field-readonly-and-disabled")
public class NumberFieldReadonlyAndDisabled extends FormLayout {

    public NumberFieldReadonlyAndDisabled() {
        setAutoResponsive(true);
        setAutoRows(true);

        // tag::snippet[]
        NumberField readonlyField = new NumberField();
        readonlyField.setReadOnly(true);
        readonlyField.setLabel("Read-only");
        readonlyField.setValue(200.0);

        NumberField disabledField = new NumberField();
        disabledField.setEnabled(false);
        disabledField.setLabel("Disabled");
        // end::snippet[]
        add(readonlyField, disabledField);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<NumberFieldReadonlyAndDisabled> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/numberfield/number-field-readonly-and-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/form-layout';
import '@vaadin/number-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('number-field-readonly-and-disabled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-form-layout auto-responsive auto-rows>
        <!-- tag::snippet[] -->
        <vaadin-number-field readonly label="Read-only" value="200"></vaadin-number-field>

        <vaadin-number-field disabled label="Disabled"></vaadin-number-field>
        <!-- end::snippet[] -->
      </vaadin-form-layout>
    `;
  }
}
